### DISCUSSION TITLE
Create separate authentication API with no Minecraft dependency

### DISCUSSION THREAD

**Discussion by triphora** [axieum/authme#31] [Ideas]
Hi,
Currently trying to implement CoolCrabs/brachyura#6 and didn't even know where to start with the Microsoft authentication. I was happy to find that AuthMe has an API, but less so to discover that it relies on a few things from Minecraft itself. Would it be possible to release a separate API with no Minecraft dependency, or maybe just remove the Minecraft dependency from the existing API?
Thanks!

**Comment by axieum** [comment]
Hi @wafflecoffee, 👋

I'm happy to take a look to see where I use Minecraft and whether I could in-house some of its functions.

Do you think the dependency tree of the published package may pose issues too though? e.g. if I publish this to Maven, can you exclude the dependency of Fabric API etc. so it isn't pulled.

Thanks,
Ax.

**Reply by triphora** [reply]
I'm not entirely sure if I can, because I've never used the Maven build tool before this, but I can try 😅

**Reply by axieum** [reply]
I fear that there may be a whole lot of extra dependencies that would need to be pulled. I use some of Minecraft's dependencies transitively, e.g. Apache HTTP.

I also construct and return a valid Minecraft session instance in that API too - which wouldn't be possible without Minecraft.

I'm more than happy for you to copy and adjust my API classes to fit your use case in your codebase. Just remember to credit me somewhere like in a `@author` Javadoc tag so anyone passing by can contact me if they need clarification on anything I had written.

**Reply by triphora** [reply]
Just wanted to say that I forgot about this for a while but that someone did eventually create one :D
https://github.com/HyCraftHD/Minecraft-Authenticator
